blazor/treeview/expand-collaspe-actions.md

```diff
@@ -0,0 +1,878 @@
+</SfTreeView>
+@code {
+    public string[] ExpandedNodes = new string[] { "1" };
+    public class MusicAlbum
+    {
+        public string Id { get; set; }
+        public string ParentId { get; set; }
+        public string Name { get; set; }
+        public bool Expanded { get; set; }
+        public bool? IsChecked { get; set; }
+        public bool HasChild { get; set; }
+    }
+    List<MusicAlbum> Albums = new List<MusicAlbum>();
+    protected override void OnInitialized()
+    {
+        base.OnInitialized();
+        Albums.Add(new MusicAlbum
+            {
+                Id = "1",
+                Name = "Discover Music",
+                HasChild = true,
+            });
+        Albums.Add(new MusicAlbum
+            {
+                Id = "2",
+                ParentId = "1",
+                Name = "Hot Singles"
+            });
+        Albums.Add(new MusicAlbum
+            {
+                Id = "3",
+                ParentId = "1",
+                Name = "Rising Artists"
+            });
+        Albums.Add(new MusicAlbum
+            {
+                Id = "4",
+                ParentId = "1",
+                Name = "Live Music"
+            });
+        Albums.Add(new MusicAlbum
+            {
+                Id = "04",
+                HasChild = true,
+                Name = "MP3 Albums"
+            });
+        Albums.Add(new MusicAlbum
+            {
+                Id = "5",
+                ParentId = "04",
+                Name = "Rock",
+                IsChecked = true
+            });
+        Albums.Add(new MusicAlbum
+            {
+                Id = "6",
+                Name = "Gospel",
+                ParentId = "04",
+            });
+        Albums.Add(new MusicAlbum
+            {
+                Id = "7",
+                ParentId = "04",
+                Name = "Latin Music"
+            });
+        Albums.Add(new MusicAlbum
+            {
+                Id = "8",
+                ParentId = "04",
+                Name = "Jazz"
+            });
+    }
+}
+
+```
+
+## Get expanded nodes
+
+The Blazor TreeView component provides the capability to expand specific nodes during initialization through the use of the [ExpandedNodes](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Navigations.SfTreeView-1.html#Syncfusion_Blazor_Navigations_SfTreeView_1_ExpandedNodes) property. Additionally, the component offers the [GetTreeData](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Navigations.SfTreeView-1.html#Syncfusion_Blazor_Navigations_SfTreeView_1_GetTreeData_System_String_) method, which allows for retrieval of the updated data source of the TreeView. By passing the ExpandedNodes of specific TreeView nodes as arguments to this method, the updated data source of only those nodes will be returned. If no arguments are passed, the entire updated data source of the TreeView will be returned.

```

Comment:
GetTreeData is not suggested model

Reply:
The GetTreeData method is used to show the expanded nodes' details example Id and name, sir.